tools/tdbg/commands.go

```diff
@@ -0,0 +1,643 @@
+			}
+		}
+
+		fmt.Printf("History service address: %s\n", resp.GetHistoryAddr())
+		fmt.Printf("Shard Id: %s\n", resp.GetShardId())
+	}
+	return nil
+}
+
+func describeMutableState(c *cli.Context) (*adminservice.DescribeMutableStateResponse, error) {
+	adminClient := cFactory.AdminClient(c)
+
+	namespace, err := getRequiredGlobalOption(c, FlagNamespace)
+	if err != nil {
+		return nil, err
+	}
+	wid := c.String(FlagWorkflowID)
+	rid := c.String(FlagRunID)
+
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	resp, err := adminClient.DescribeMutableState(ctx, &adminservice.DescribeMutableStateRequest{
+		Namespace: namespace,
+		Execution: &commonpb.WorkflowExecution{
+			WorkflowId: wid,
+			RunId:      rid,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Get workflow mutableState failed: %s", err)
+	}
+	return resp, nil
+}
+
+// AdminDeleteWorkflow delete a workflow execution from Cassandra and visibility document from Elasticsearch.
+func AdminDeleteWorkflow(c *cli.Context) error {
+	return fmt.Errorf("Not implemented")
+}
+
+func adminDeleteVisibilityDocument(c *cli.Context, namespaceID string) error {
+	if !c.IsSet(FlagElasticsearchIndex) {
+		prompt("Elasticsearch index name is not specified. Continue without visibility document deletion?", c.Bool(FlagAutoConfirm))
+	}
+
+	indexName := c.String(FlagElasticsearchIndex)
+	esClient, err := newESClient(c)
+	if err != nil {
+		return err
+	}
+
+	query := elastic.NewBoolQuery().
+		Filter(
+			elastic.NewTermQuery(searchattribute.NamespaceID, namespaceID),
+			elastic.NewTermQuery(searchattribute.WorkflowID, c.String(FlagWorkflowID)))
+	if c.IsSet(FlagRunID) {
+		query = query.Filter(elastic.NewTermQuery(searchattribute.RunID, c.String(FlagRunID)))
+	}
+
+	searchParams := &esclient.SearchParameters{
+		Index:    indexName,
+		Query:    query,
+		PageSize: 10000,
+	}
+	searchResult, err := esClient.Search(context.Background(), searchParams)
+	if err != nil {
+		if c.Bool(FlagSkipErrorMode) {
+			fmt.Println("Unable to search for visibility documents from Elasticsearch:", err)
+		} else {
+			return fmt.Errorf("Unable to search for visibility documents from Elasticsearch: %s", err)
+		}
+	}
+	fmt.Println("Found", len(searchResult.Hits.Hits), "visibility documents.")
+	for _, searchHit := range searchResult.Hits.Hits {
+		err := esClient.Delete(context.Background(), indexName, searchHit.Id, math.MaxInt64)
+		if err != nil {
+			if c.Bool(FlagSkipErrorMode) {
+				fmt.Println("Unable to delete visibility document from Elasticsearch:", err)
+			} else {
+				return fmt.Errorf("Unable to delete visibility document from Elasticsearch: %s", err)
+			}
+		} else {
+			fmt.Println("Visibility document", searchHit.Id, "deleted successfully.")
+		}
+	}
+	return nil
+}
+
+func readOneRow(query gocql.Query) (map[string]interface{}, error) {
+	result := make(map[string]interface{})
+	err := query.MapScan(result)
+	return result, err
+}
+
+func connectToCassandra(c *cli.Context) (gocql.Session, error) {
```

Comment:
This function should not be used anymore and needs to be removed.

Reply:
done